app/models/spree_avatax/shared.rb

```diff
@@ -37,6 +55,12 @@ def tax_svc
       })
     end
 
+    def with_timeout
+      Timeout.timeout(SpreeAvatax::Config.timeout, SpreeAvatax::AvataxTimeout) do
```

Comment:
We should mention that we looked at the Avatax gem to make sure they don't use any sort of connection pooling or the like which might cause problems with the use of `Timeout.timeout`.